### PR TITLE
Update wasmtime

### PR DESCRIPTION
**Pull request by NikVolf**
this updates wasmtime to *almost* current bytecode alliance master

Should probably ice-d until they release 0.16

### CODE REVIEW

**Comment by arkpar** [comment]
Some tests need to be updated

File: client/executor/wasmtime/src/host.rs [hunk 1]
```diff
@@ -117,7 +117,7 @@ impl<'a> SandboxCapabilities for HostContext<'a> {
 					return Err("Supervisor function returned unexpected result!".into());
 				}
 			}
-			Err(err) => Err(err.message().to_string().into()),
+			Err(err) => Err(format!("{}", err).into()),
```
Comment (bkchr) [inline client/executor/wasmtime/src/host.rs:120]:
```suggestion
			Err(err) => Err(err.to_string().into()),
```

**Comment by bkchr** [comment]
`0.16` was released.

**Comment by arkpar** [comment]
Let's pull upstream changes into paritytech/wasmtime once more and publish it
@gnunicorn @NikVolf

**Comment by NikVolf** [comment]
This (merged right before 0.16 release) unfortunately made our fork hardly maintainable
https://github.com/bytecodealliance/wasmtime/pull/1624

I'll update until that point and try to publish that

**Comment by NikVolf** [comment]
Once this green, we reset `wasmtime/master` to `wasmtime/master-candidate`

Then our crates `substrate-wasmtime`, `substrate-wasmtime-runtime` needs to be published @gnunicorn 

Then this can be merged

**Comment by bkchr** [comment]
@NikVolf your patches are not applied. You should at least test your stuff locally.

@gnunicorn please invite the paritytech/team for the `substrate-wasmtime*` crates on `crates.io`

**Comment by NikVolf** [comment]
> You should at least test your stuff locally.

Kusama is syncing fine

**Comment by bkchr** [comment]
I meant the compilation of this branch^^ The stable test is still not working.

**Comment by NikVolf** [comment]
As discussed, the warning is irrelevant, patch is being used

We just need to publish our variants of `wasmtime` (`substrate-wasmtime-*`) crates @gnunicorn

**Comment by gnunicorn** [comment]
`substrate-wasmtime = "0.16.0-threadsafe.1"` is published (and all `substrate-wasmtime-*` crates are also owned by core-devs, now).

**Pull request merged**